From cced635e9555d0563f628a9828f0e58823598aaa Mon Sep 17 00:00:00 2001
From: John Sanda <jsanda@redhat.com>
Date: Sun, 10 Oct 2010 11:16:42 -0400
Subject: [PATCH] Initial commit for CreateResourceHistoryCriteria.java

This commit also includes some initial support for testing entities with
unitils.
---
 modules/core/domain/pom.xml                        |   43 +++++++++++++
 .../criteria/CreateResourceHistoryCriteria.java    |   67 ++++++++++++++++++++
 .../CreateResourceHistoryCriteriaTest.java         |   15 +++++
 .../criteria/CreateResourceHistoryCriteriaTest.xml |    9 +++
 .../domain/src/test/resources/unitils.properties   |   27 ++++++++
 5 files changed, 161 insertions(+), 0 deletions(-)
 create mode 100644 modules/core/domain/src/main/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteria.java
 create mode 100644 modules/core/domain/src/test/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.java
 create mode 100644 modules/core/domain/src/test/resources/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.xml
 create mode 100644 modules/core/domain/src/test/resources/unitils.properties

diff --git a/modules/core/domain/pom.xml b/modules/core/domain/pom.xml
index 0806490..3356c0d 100644
--- a/modules/core/domain/pom.xml
+++ b/modules/core/domain/pom.xml
@@ -240,6 +240,49 @@
            <version>2.0</version>
        </dependency>-->
 
+        <dependency>
+            <groupId>org.unitils</groupId>
+            <artifactId>unitils-dbunit</artifactId>
+            <version>3.1</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.unitils</groupId>
+            <artifactId>unitils-orm</artifactId>
+            <version>3.1</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.unitils</groupId>
+            <artifactId>unitils-dbmaintainer</artifactId>
+            <version>3.1</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.unitils</groupId>
+            <artifactId>unitils-easymock</artifactId>
+            <version>3.1</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.unitils</groupId>
+            <artifactId>unitils-testng</artifactId>
+            <version>3.1</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.unitils</groupId>
+            <artifactId>unitils-inject</artifactId>
+            <version>3.1</version>
+            <scope>test</scope>
+        </dependency>
+       
+
    </dependencies>
 
    <build>
diff --git a/modules/core/domain/src/main/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteria.java b/modules/core/domain/src/main/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteria.java
new file mode 100644
index 0000000..1cf91ce
--- /dev/null
+++ b/modules/core/domain/src/main/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteria.java
@@ -0,0 +1,67 @@
+package org.rhq.core.domain.criteria;
+
+import org.rhq.core.domain.resource.CreateResourceHistory;
+import org.rhq.core.domain.resource.CreateResourceStatus;
+
+public class CreateResourceHistoryCriteria extends TaggedCriteria {
+
+    private Integer filterId;
+    private Integer filterParentResourceId; // needs overrides
+    private String filterParentResourceName; // needs overrides
+    private Integer filterParentResourceTypeId; // needs overrides
+    private String filterCreateResourceName;
+    private String filterSubjectName;
+    private CreateResourceStatus filterStatus;
+    private String filterPackageVersion;
+    private String filterPackageFileName;
+
+    public CreateResourceHistoryCriteria() {
+        filterOverrides.put("parentResourceId", "parentResource.id = ?");
+        filterOverrides.put("parentResourceName", "parentResource.name like ?");
+        filterOverrides.put("parentResourceTypeId", "parentResource.resourceType.id = ?");
+        filterOverrides.put("packageVersion", "installedPackage.packageVersion.version = ?");
+        filterOverrides.put("packageFileName", "installedPackage.packageVersion.fileName = ?");
+    }
+
+    @Override
+    public Class<CreateResourceHistory> getPersistentClass() {
+        return CreateResourceHistory.class;
+    }
+
+    public void addFilterId(Integer id) {
+        filterId = id;
+    }
+
+    public void addFilterParentResourceId(Integer filterParentResourceId) {
+        this.filterParentResourceId = filterParentResourceId;
+    }
+
+    public void addFilterParentResourceName(String filterParentResourceName) {
+        this.filterParentResourceName = filterParentResourceName;
+    }
+
+    public void addFilterParentResourceTypeId(int filterParentResourceTypeId) {
+        this.filterParentResourceTypeId = filterParentResourceTypeId;
+    }
+
+    public void addFilterCreateResourceName(String createResourceName) {
+        filterCreateResourceName = createResourceName;
+    }
+
+    public void addFilterSubjectName(String subjectName) {
+        filterSubjectName = subjectName;
+    }
+
+    public void addFilterStatus(CreateResourceStatus status) {
+        filterStatus = status;
+    }
+
+    public void addFilterPackageVersion(String version) {
+        filterPackageVersion = version;
+    }
+
+    public void addFilterPackageFileName(String fileName) {
+        filterPackageFileName = fileName;
+    }
+
+}
diff --git a/modules/core/domain/src/test/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.java b/modules/core/domain/src/test/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.java
new file mode 100644
index 0000000..c39766e
--- /dev/null
+++ b/modules/core/domain/src/test/java/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.java
@@ -0,0 +1,15 @@
+package org.rhq.core.domain.criteria;
+
+import org.testng.annotations.Test;
+import org.unitils.UnitilsTestNG;
+import org.unitils.dbunit.annotation.DataSet;
+
+@DataSet
+public class CreateResourceHistoryCriteriaTest extends UnitilsTestNG {
+
+    @Test
+    public void testUnitils() {
+        System.out.println("It Worked!");
+    }
+
+}
diff --git a/modules/core/domain/src/test/resources/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.xml b/modules/core/domain/src/test/resources/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.xml
new file mode 100644
index 0000000..549b01d
--- /dev/null
+++ b/modules/core/domain/src/test/resources/org/rhq/core/domain/criteria/CreateResourceHistoryCriteriaTest.xml
@@ -0,0 +1,9 @@
+<dataset>
+    <rhq_subject id="-1"
+                 name="test"
+                 first_name="test"
+                 last_name="user"
+                 email_address="test@localhost"
+                 factive="true"
+                 fsystem="false"/>
+</dataset>
\ No newline at end of file
diff --git a/modules/core/domain/src/test/resources/unitils.properties b/modules/core/domain/src/test/resources/unitils.properties
new file mode 100644
index 0000000..9515528
--- /dev/null
+++ b/modules/core/domain/src/test/resources/unitils.properties
@@ -0,0 +1,27 @@
+# Defaults and other keys with explanations can be found there: http://unitils.org/unitils-default.properties
+unitils.module.database.enabled=true
+unitils.module.dbunit.enabled=true
+unitils.module.hibernate.enabled=false
+unitils.module.mock.enabled=false
+unitils.module.easymock.enabled=false
+unitils.module.inject.enabled=true
+unitils.module.spring.enabled=false
+unitils.module.jpa.enabled=true
+
+database.driverClassName=org.postgresql.Driver
+database.url=jdbc:postgresql://127.0.0.1:5432/rhq
+
+database.dialect=postgresql
+database.userName=rhqadmin
+database.password=rhqadmin
+database.schemaNames=public
+
+org.unitils.core.dbsupport.DbSupport.implClassName.postgresql=org.unitils.core.dbsupport.PostgreSqlDbSupport
+org.dbunit.dataset.datatype.IDataTypeFactory.implClassName.postgresql=org.dbunit.dataset.datatype.DefaultDataTypeFactory
+
+DbUnitModule.DataSet.loadStrategy.default=org.unitils.dbunit.datasetloadstrategy.impl.RefreshLoadStrategy
+
+
+updateDataBaseSchema.enabled=false
+
+jpa.persistenceProvider=hibernate
-- 
1.7.2.3

